.githooks/pre-commit

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+if [ "$SEARXNG_PRECOMMIT" = "true" ]; then
+    staged_files=$(git diff --cached --name-only)
+    python_files=$(echo "$staged_files" | grep '\.py$')
+    # Only attempt to run the formatter if Python files were changes
+    if [ -n "$python_files" ]; then
+        make format.python
```

Comment:
I'd like to make this more performant by passing in the list of generated files, but this is not an option at this point of time. I can definitely include this in a follow up PR.